src/baal/transformerswrapper.py

```diff
@@ -0,0 +1,76 @@
+import torch
+from typing import Optional, List, Sequence
+import numpy as np
+from transformers import Trainer
+
+from baal.utils.array_utils import stack_in_memory
+from baal.utils.iterutils import map_on_tensor
+
+
+class BaalHuggingFaceTrainer(Trainer):
+
+
+    def predict_on_dataset_generator(self,
+                                     dataset,
+                                     iterations: int = 1,
+                                     half: bool = False,
+                                     ignore_keys: Optional[List[str]] = None):
+
+        dataloader = self.get_eval_dataloader(dataset)
+
+        model = self.model
+
+        # multi-gpu eval
+        if self.args.n_gpu > 1:
+            model = torch.nn.DataParallel(model)
+        # Note: in torch.distributed mode, there's no point in wrapping the model
+        # inside a DistributedDataParallel as we'll be under `no_grad` anyways.
+
+        model.eval()
+        # with torch.no_grad():
+        #     if cuda:
+        #         data = to_cuda(data)
+        for step, inputs in enumerate(dataloader):
+            inputs = map_on_tensor(lambda d: stack_in_memory(d, iterations), inputs)
```

Comment:
It would be useful to add comments to this block of code, to understand why we are doing it as it is not immediately obvious